python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -652,6 +651,56 @@ def train_step(self, graph, feat, tmp, **kwargs):
         """
         return super().train_step(graph, feat, tmp=tmp, **kwargs)
 
+    def train_step_node(self, nodes, graph, feat, tmp, **kwargs):
+        r"""Compute the loss of the explanation network for node classification
+
+        Parameters
+        ----------
+        nodes : dict[str, interable[int]]
+            A dict mapping node types (keys) to an iterable set of node ids (values).
+        graph : DGLGraph
+            Input heterogeneous graph.
+        feat : dict[str, Tensor]
+            A dict mapping node types (keys) to feature tensors (values).
+            The input features are of shape :math:`(N_t, D_t)`. :math:`N_t` is the
+            number of nodes for node type :math:`t`, and :math:`D_t` is the feature
+            size for node type :math:`t`
+        tmp : float
+            The temperature parameter fed to the sampling procedure.
+        kwargs : dict
+            Additional arguments passed to the GNN model.
+
+        Returns
+        -------
+        Tensor
+            A scalar tensor representing the loss.
+        """
+        assert (
+            not self.graph_explanation
+        ), '"explain_graph" must be False in initializing the module.'
```

Comment:
So it seems like when `graph_explanation` attribute is True, then we can only use `explain_graph` method.  Otherwise, we can only use `explain_node` method.  I didn't see a corresponding assertion in `explain_graph` though.  Same for `train_step`.

@frozenbugs do you think we should split `graph_explanation` argument into two classes, like `NodeExplainer` and `GraphExplainer`?

Reply:
2 questions to be answer:
- are NodeExplainer and GraphExplainer mutually exclusive?
- In the future will there be another Explainer like edge explainer?

If "no" to Q1 or "yes" to Q2, we should split it to 2 classes, otherwise, the currently api is fine-ish.

Reply:
SGTM.